code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm

```diff
@@ -153,12 +198,19 @@
 	if(!istype(target))
 		return
 
-	to_chat(X, "<span class='xenonotice'>We begin building up acid.</span>")
+	if(X.ammo.type == /datum/ammo/xeno/boiler_gas/corrosive)
+		if(X.xeno_caste.corrosive_ammo <= 0)
+			to_chat(X, "<span class='warning'>We have no corrosive globules available.</span>")
+			return
+	else

```

Comment:
There's a return, the `else` is not needed.

Reply:
Actually, this shouldn't be in the caste.
See, castes are singletons. Only one of each type exists for all xenos. They store variables like this.
Put these variables in the action datum itself.